### PR TITLE
WT-12475 remove indexes tests and ex

### PR DESCRIPTION
**Pull request by ruby-oujo**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -5882,8 +5880,6 @@ buildvariants:
       distros: amazon2-arm64-large
     - name: format-smoke-test
     - name: format-stress-pull-request-test
-    - name: linux-directio
```
Comment (ruby-oujo) [inline test/evergreen.yml:5885]: This keeps slipping back into the changes, I think when merging develop this goes back in without merge conflicts :/

File: examples/c/ex_col_store.c
Comment (ruby-oujo) [inline examples/c/ex_col_store.c:1]: Removing the usage of indexes affects most of this code; I spoke to @Sean04 about the effort to rewrite this and was probably not worth it.

File: test/csuite/wt2909_checkpoint_integrity/main.c [hunk 2]
```diff
@@ -52,13 +52,9 @@
  * given N may give different results on different runs. But approximate optimal N can be
  * determined, allowing a series of additional tests clustered around this N.
  *
- * The data is stored in two tables, one having indices. Both tables have the same keys and are
```
Comment (tod-johnson-mongodb) [inline test/csuite/wt2909_checkpoint_integrity/main.c:55]: Please rewrite to explain the two tables without mentioning indexes.

* The data is stored in two tables. Both tables have the same keys and are
* updated with the same key in a single transaction.
*

File: test/suite/test_alter01.py [hunk 3]
```diff
@@ -123,20 +122,17 @@ def test_alter01_access(self):
             cache_param = 'cache_resident=false'
 
         cgparam = ''
-        if self.use_cg or self.use_index:
+        if self.use_cg:
             cgparam = 'columns=(k,v),'
         if self.use_cg:
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter01.py:127]: New lines 126-128 can be changed to: cgparam = 'columns=(k,v),colgroups=(g0),"

File: test/suite/test_alter01.py [hunk 4]
```diff
@@ -167,7 +163,7 @@ def test_alter01_access(self):
                     self.session.alter(uri, alter_param)
                     if self.reopen:
                         self.reopen_conn()
-                    special = self.use_cg or self.use_index
+                    special = self.use_cg
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter01.py:166]: Delete new line 166.

File: test/suite/test_alter01.py [hunk 5]
```diff
@@ -167,7 +163,7 @@ def test_alter01_access(self):
...
                     if not special:
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter01.py:167]: Change to if not self.use_cg:

File: test/suite/test_alter02.py [hunk 6]
```diff
@@ -165,20 +164,17 @@ def test_alter02_log(self):
         complex_params += '%s,' % log_param
 
         cgparam = ''
-        if self.use_cg or self.use_index:
+        if self.use_cg:
             cgparam = 'columns=(k,v),'
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter02.py:168]: Combine cgparam lines into one line.

File: test/suite/test_alter02.py [hunk 7]
```diff
@@ -207,7 +203,7 @@ def test_alter02_log(self):
             log_str = 'log=(enabled=false)'
             table_logged = 0
         alter_param = '%s' % log_str
-        special = self.use_cg or self.use_index
+        special = self.use_cg
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter02.py:206]: Delete special and replace with self.use_cg

File: test/suite/test_alter04.py [hunk 8]
```diff
@@ -106,20 +105,17 @@ def test_alter04_cache(self):
             new_param = '%s=0' % self.setting
 
         cgparam = ''
-        if self.use_cg or self.use_index:
+        if self.use_cg:
             cgparam = 'columns=(k,v),'
         if self.use_cg:
             cgparam += 'colgroups=(g0),'
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter04.py:111]: Combine cgparam lines into one line.

File: test/suite/test_alter04.py [hunk 9]
```diff
@@ -138,7 +134,7 @@ def test_alter04_cache(self):
             if self.reopen:
                 self.reopen_conn()
 
-            special = self.use_cg or self.use_index
+            special = self.use_cg
```
Comment (tod-johnson-mongodb) [inline test/suite/test_alter04.py:137]: Delete special and replace with self.use_cg.

File: test/suite/test_backup28.py [hunk 10]
```diff
@@ -59,15 +58,11 @@ def test_backup28(self):
...
         if (self.pfx != "lsm:" and self.pfx != "file:"):
-            # Add in column group and index tables.
+            # Add in column group tables.
             colgroup_param = 'columns=(v),'
             suburi = 'colgroup:table0:g0'
             self.session.create(suburi, colgroup_param)
 
-            suburi = 'index:table0:i0'
-            self.session.create(suburi, cgparam)
-            self.session.checkpoint()
```
Comment (tod-johnson-mongodb) [inline test/suite/test_backup28.py:69]: Why is old line 69 self.session.checkpoint() deleted?

File: test/suite/test_cursor_bound01.py [hunk 11]
```diff
@@ -58,7 +57,7 @@ def test_bound_api(self):
 
         uri = self.uri + self.file_name
         create_params = 'value_format={},key_format={}'.format(self.value_format, self.key_format)
-        if self.use_index or self.use_colgroup:
+        if self.use_colgroup:
             create_params += ",columns=(k,v)"
         if self.use_colgroup:
```
Comment (tod-johnson-mongodb) [inline test/suite/test_cursor_bound01.py:62]: Combine create_params lines into one line.

File: test/suite/test_cursor_compare.py [hunk 12]
```diff
@@ -121,16 +110,6 @@ def test_cursor_comparison(self):
             ix0_1.prev()
             self.assertLess(ix0_0.compare(ix0_1), 0)
             self.assertGreater(ix0_1.compare(ix0_0), 0)
-            # Main table vs. index not allowed
-            msg = '/must reference the same object/'
-            self.assertRaisesWithMessage(
-                wiredtiger.WiredTigerError, lambda: c1.compare(ix0_0), msg)
-            # Two unrelated indices not allowed
-            self.assertRaisesWithMessage(
```
Comment (tod-johnson-mongodb) [inline test/suite/test_cursor_compare.py:129]: ix_0 always == None. Delete new lines 107-112 and new lines 74-77.

File: test/suite/test_cursor_compare.py [hunk 13]
```diff
@@ -209,16 +178,6 @@ def test_cursor_equality(self):
             ix0_1.reset()
             ix0_1.prev()
             self.assertFalse(ix0_0.equals(ix0_1))
-            # Main table vs. index not allowed
-            msg = '/must reference the same object/'
-            self.assertRaisesWithMessage(
```
Comment (tod-johnson-mongodb) [inline test/suite/test_cursor_compare.py:214]: ix0_0 always == None. Delete new lines 176-180 and 143-146.

File: test/suite/test_metadata_cursor04.py [hunk 14]
```diff
@@ -53,18 +53,16 @@ def check_meta(self, uri, check_log):
             self.pr(uri + " SUCCESS")
         self.assertTrue(log_false != -1)
 
-    # Test a complex table with column groups and an index.
+    # Test a complex table with column groups..
```
Comment (tod-johnson-mongodb) [inline test/suite/test_metadata_cursor04.py:56]: Change ".." to "."

File: test/suite/test_pack.py [hunk 15]
```diff
@@ -71,12 +67,6 @@ def check(self, fmt, *v):
             got = [got]
         self.assertEquals(got, v)
 
-        forw_idx.set_key(*v)
-        self.assertEquals(forw_idx.search(), 0)
-        self.assertEquals(forw_idx.get_value(), 1234)
-        forw.close()
```
Comment (tod-johnson-mongodb) [inline test/suite/test_pack.py:77]: Do not delete old line 77.

File: tools/backup_analysis.py [hunk 16]
```diff
@@ -93,8 +93,7 @@ def older_dir(dir1, dir2):
 # is kept to analyze the state of the system.
 def compute_type(filename, filemeta):
     # Figure out the type of file it is based on the name and metadata.
-    # For collections and indexes, if logging is disabled, then they are replicated.
-    #   if logging is enabled on an index, it is a local table.
+    # For collections if logging is disabled, then it is replicated.
```
Comment (tod-johnson-mongodb) [inline tools/backup_analysis.py:96]: Change "they is replicated" to "they are replicated" since collections is plural.

**Review by tod-johnson-mongodb — CHANGES_REQUESTED** [review]
Some changes needed.
Also does not pass some tests.

**Comment by korteland** [comment]
Closing this out as the project is on ice for Q3.

**Pull request closed without merging**